builder/includes/datastream.form.inc

```diff
@@ -184,3 +184,74 @@ function xml_form_builder_datastream_form_metadata_form_submit(array $form, arra
   xml_form_builder_update_object($object, $association, $document, $label);
   $form_state['redirect'] = "islandora/object/{$object->id}";
 }
+
+/**
+ * Form for selecting which metadata datastream to create.
+ *
+ * @param array $form
+ *   An array representing a form within Drupal.
+ * @param array $form_state
+ *   An array containing the Drupal form state.
+ * @param AbstractObject $object
+ *   Object to add the new metadata datastream to.
+ *
+ * @return array
+ *   An array containing the form to be rendered.
+ */
+function xml_form_builder_create_metadata_form($form, &$form_state, $object) {
+  form_load_include($form_state, 'inc', 'xml_form_builder', 'includes/datastream.form');
+  form_load_include($form_state, 'inc', 'xml_form_builder', 'includes/associations');
+  form_load_include($form_state, 'inc', 'islandora', 'includes/utilities');
+
+  $form_state['object_id'] = $object->id;
+  // Find all possible datastreams that don't already exist.
+  $datastreams = array();
+  foreach (islandora_get_datastreams_requirements($object) as $dsid => $datastream) {
+    if (!isset($object[$dsid])) {
+      $datastreams[] = $dsid;
+    }
+  }
+  // Find datastreams that associations exist for.
+  if (!empty($datastreams)) {
```

Comment:
If there no applicable datastreams we are still creating an empty fieldset with not content, perhaps we can bubble up the check to `xml_form_builder_add_datastream_page` to not create the fieldset?